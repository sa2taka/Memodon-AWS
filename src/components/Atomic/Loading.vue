<template>
  <div :class="[isComplete? 'complete' : 'spinner']">
  	<svg viewBox="0 0 66 66" xmlns="http://www.w3.org/2000/svg">
  		<rect v-if="isComplete" width="66" height="66" fill="none"/>
  		<polyline v-if="isComplete" points="7 43 26 58 60 18" fill="none" stroke="#fff" stroke-linecap="square" stroke-miterlimit="10" stroke-width="6"/>
  		<circle v-else class="length" fill="none" stroke-width="8" stroke-linecap="round" cx="33" cy="33" r="28"></circle>
  	</svg>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';
import theme from '@/store/modules/theme';

@Component
export default class Loading extends Vue {
  @Prop()
  public isComplete: boolean = false;

  public get circle(): string {
    return this.drawCircle(26, 33, 33).join(' ');
  }

  private drawCircle(r: number, centerX: number, centerY: number) {
    return [
      `M${centerX - r},${centerY}`,
      `A${r},${r},0,1,0,${centerX + r},${centerY}`,
      `A${r},${r},0,1,0,${centerX - r},${centerY}`,
    ];
  }
}
</script>

<style lang="scss" scoped>
*, *:before, *:after {
	box-sizing: border-box;
	position: relative;
}

svg {
	display: block;
	
	margin: 0; 
	padding: 0;
}

.spinner, .complete {
	width: 66px; height: 66px;
}

.spinner {
	animation: contanim 2.5s linear infinite;
}

$color: #84EBBD;
$d: 175.6449737548828;

svg {
	width: 100%; 
	height: 100%;
	
	left: 0; 
	top: 0;
	position: absolute;
	
	/*transform: rotate(-90deg);*/

	 circle {
		stroke: $color;
		stroke-dasharray: 1, 300;
		stroke-dashoffset: 0;
			
		animation: strokeanim 2.0s 0.2s ease infinite;
		transform-origin: center center;
	}
}

@keyframes strokeanim {
	0% {
    stroke-dasharray: 1, 300;
    stroke-dashoffset: 0;
  }
  50% {
    stroke-dasharray: 120, 300;
    stroke-dashoffset: -$d / 3;
  }
  100% {
    stroke-dasharray: 120, 300;
    stroke-dashoffset: -$d;
  }
}

@keyframes contanim {
	100% {
		transform: rotate(360deg)
	}
}
</style>
