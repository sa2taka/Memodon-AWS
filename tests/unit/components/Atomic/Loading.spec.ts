import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import Loading from '@/components/Atomic/Loading.vue';

const localVue = createLocalVue()

localVue.use(Vuetify);

describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(Loading, { localVue });
    expect(wrapper.element).toMatchSnapshot();
  });
  
  test('become checkmark when process complete', () => {
    const wrapper = shallowMount(Loading, {
      propsData: {
        isComplete: true,  
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  
  test('become cross when process error', () => {
    const wrapper = shallowMount(Loading, {
      propsData: {
        isError: true,  
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
