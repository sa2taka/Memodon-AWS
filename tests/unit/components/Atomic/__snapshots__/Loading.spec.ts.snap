// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component become checkmark when process complete 1`] = `
<div
  class="complete"
  style="width: 66px; height: 66px;"
>
  <svg
    class="icon"
    name="checkmark"
    viewBox="0 0 66 66"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="none"
      height="66"
      width="66"
    />
     
    <polyline
      fill="none"
      points="4 41 25 58 62 17"
      stroke="#009688"
      stroke-linecap="square"
      stroke-miterlimit="10"
      stroke-width="7.333333333333333"
    />
  </svg>
   
  <!---->
   
  <!---->
</div>
`;

exports[`Component become cross when process error 1`] = `
<div
  class="error-icon"
  style="width: 66px; height: 66px;"
>
  <!---->
   
  <svg
    class="icon"
    name="error"
    viewBox="0 0 66 66"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="#f44336"
      stroke="none"
      transform="scale(0.00515625, 0.00515625)"
    >
      <path
        d="M1545 12784 c-85 -19 -167 -51 -243 -95 -69 -41 -1089 -1049 -1157
              -1144 -101 -141 -140 -263 -140 -440 0 -169 36 -293 125 -427 29 -43 705 -726
              2149 -2170 l2106 -2108 -2111 -2112 c-1356 -1358 -2124 -2133 -2147 -2169 -88
              -137 -121 -249 -121 -419 -1 -181 37 -302 139 -445 68 -95 1088 -1103 1157
              -1144 273 -159 604 -143 853 42 22 17 986 976 2143 2131 l2102 2101 2103
              -2101 c1156 -1155 2120 -2114 2142 -2131 69 -51 130 -82 224 -113 208 -70 431
              -44 629 71 69 41 1089 1049 1157 1144 101 141 140 263 140 440 0 166 -36 290
              -121 422 -25 39 -746 767 -2148 2171 l-2111 2112 2107 2108 c2207 2208 2162
              2161 2219 2303 75 187 77 392 4 572 -53 132 -74 157 -615 700 -289 291 -552
              548 -585 572 -141 101 -263 140 -440 140 -166 0 -289 -35 -420 -120 -41 -26
              -724 -702 -2172 -2149 l-2113 -2111 -2112 2111 c-1454 1452 -2132 2123 -2173
              2150 -64 41 -149 78 -230 101 -79 22 -258 26 -340 7z"
      />
    </g>
  </svg>
   
  <!---->
</div>
`;

exports[`Component is a Vue instance 1`] = `
<div
  class="spinner"
  style="width: 66px; height: 66px;"
>
  <!---->
   
  <!---->
   
  <svg
    class="icon"
    name="loading"
    viewBox="0 0 66 66"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      class="length"
      cx="33"
      cy="33"
      fill="none"
      r="28"
      stroke="#009688"
      stroke-linecap="round"
      stroke-width="8"
    />
  </svg>
</div>
`;
